tuf/tuf.go

```diff
@@ -143,13 +143,16 @@ func (tr *Repo) RemoveBaseKeys(role string, keyIDs ...string) error {
 		}
 	}
 
-	// remove keys no longer in use by any roles
-	for k := range toDelete {
-		delete(tr.Root.Signed.Keys, k)
-		// remove the signing key from the cryptoservice if it
-		// isn't a root key. Root keys must be kept for rotation
-		// signing
-		if role != data.CanonicalRootRole {
+	// Remove keys no longer in use by any roles, except for root keys.
+	// Root private keys must be kept in tr.cryptoService to be able to sign
+	// for rotation, and root certificates must be kept in tr.Root.SignedKeys
+	// because we are not necessarily storing them elsewhere (tuf.Repo does not
+	// depend on certs.Manager, that is an upper layer), and without storing
+	// the certificates in their x509 form we are not able to do the
+	// util.CanonicalKeyID conversion.
+	if role != data.CanonicalRootRole {
+		for k := range toDelete {
+			delete(tr.Root.Signed.Keys, k)
```

Comment:
I would greatly prefer to use the previous root.json's signature list to decide which old keys to sign with. Leaving old root keys in the list creates a special case. Anyone looking at the root.json has to make the assumption or know already that orphaned keys entries are the special case of old root keys.

Using that model, ceasing to sign with an old key (i.e. because it is years old and considered no longer relevant to rotations) is as simple as deleting if from the keystore, rather than needing an additional command to remove the orphaned key from the root.json as well.